File: job-agent-os/src/job_agent_os/schemas/interview.py
"""Interview schemas."""

from uuid import UUID

from pydantic import BaseModel, Field


class QuestionGenerateRequest(BaseModel):
    """Interview question generation request."""

    job_id: UUID
    resume_id: UUID | None = None
    question_types: list[str] = Field(default_factory=lambda: ["technical", "behavioral"])
    difficulty: str = Field(default="mixed", pattern="^(easy|medium|hard|mixed)$")
    count: int = Field(default=10, ge=1, le=50)


class InterviewQuestion(BaseModel):
    """Interview question."""

    id: int
    type: str
    category: str
    difficulty: str
    question: str
    reference_answer: str
    scoring_criteria: list[str] = Field(default_factory=list)
    follow_up: str | None = None


class InterviewQuestionSet(BaseModel):
    """Interview question set response."""

    status: str
    job_title: str | None = None
    questions: list[InterviewQuestion] = Field(default_factory=list)

File: job-agent-os/frontend/src/lib/api/interview.ts
/** Interview API */

import { api } from "./client";
import type { InterviewQuestion } from "@/types/session";

/** Re-export InterviewQuestion as GeneratedQuestion for backward compatibility. */
export type GeneratedQuestion = InterviewQuestion;

export interface QuestionGenerateRequest {
  job_id: string;
  resume_id?: string | null;
  question_types?: string[];
  difficulty?: "easy" | "medium" | "hard" | "mixed";
  count?: number;
}

export interface GenerationTaskResponse {
  task_id: string;
  status: "processing" | "completed" | "failed" | string;
  result: {
    status: string;
    job_title: string | null;
    questions: GeneratedQuestion[];
  } | null;
}

export function generateQuestions(data: QuestionGenerateRequest) {
  return api.post<{ task_id: string; status: string }>("/v1/interview/questions", data);
}

export function getGenerationResult(taskId: string) {
  return api.get<GenerationTaskResponse>(`/v1/interview/questions/${taskId}`);
}
